python/tk_multi_demo/syntax_highlighter.py

```diff
@@ -82,8 +82,8 @@ def __init__(self, document, palette):
         # Multi-line strings (expression, flag, style)
         # FIXME: The triple-quotes in these two lines will mess up the
         # syntax highlighting from this point onward
-        self.tri_single = (QtCore.QRegExp("'''"), 1, self._style("string2"))
-        self.tri_double = (QtCore.QRegExp('"""'), 2, self._style("string2"))
+        self.tri_single = (QtCore.QRegularExpression("'''"), 1, self._style("string2"))
+        self.tri_double = (QtCore.QRegularExpression('"""'), 2, self._style("string2"))
```

Comment:
This will break QT4 since `QRegularExpression` does not exist until I believe QT5. Assuming we want to deprecate but not break QT4, we would need to handle both QRegExp/QRegularExpression at the same time. The tk-multi-pythoncosole app deals with QRegExp (QT4) / QRegularExpression (QT6):
- https://github.com/shotgunsoftware/tk-multi-pythonconsole/blob/b875b8a25e9e1a30e924e5de63129383b2cf19bb/python/app/qt_importer.py#L70-L77).
- https://github.com/shotgunsoftware/tk-multi-pythonconsole/blob/b875b8a25e9e1a30e924e5de63129383b2cf19bb/python/app/syntax_highlighter.py

QT5 has both QRegExp and QRegularExpression functionality

Reply:
> This will break QT4 since `QRegularExpression` does not exist until I believe QT5.

This is not part of this PR but we do plan to break Qt4 compatibility very soon.